### PR TITLE
623 type conv address

### LINKED ISSUE #623: Modify types to ease type conversions
Modify types to ease type conversions.

- [x] Use common.Address for addresses
- [x] Use hexutil.Byte for hex payloads
- [x] Use hexutil.Big for big numbers when they are expected as hex strings
~- [ ] Use []byte for base64 payloads (like privateFrom)~

### PR DESCRIPTION
**Pull request by amkCha**
## Fixed Issue(s)

fixes #623

### CODE REVIEW

**Review by darioAnongba — APPROVED** [review]

File: pkg/types/api/deploy_tx_request.go [hunk 1]
```diff
@@ -20,7 +21,7 @@ type DeployContractParams struct {
 	GasTipCap       string                        `json:"maxPriorityFeePerGas,omitempty" example:"71500000 (wei)"`
 	AccessList      types.AccessList              `json:"accessList,omitempty" swaggertype:"array,object"`
 	TransactionType string                        `json:"transactionType,omitempty" validate:"omitempty,isTransactionType" example:"dynamic_fee" enums:"legacy,dynamic_fee"`
-	From            string                        `json:"from" validate:"omitempty,eth_addr" example:"0x1abae27a0cbfb02945720425d3b80c7e09728534"`
+	From            *ethcommon.Address             `json:"from" validate:"omitempty" example:"0x1abae27a0cbfb02945720425d3b80c7e09728534" swaggertype:"string"`
```
Comment (ggarri) [inline pkg/types/api/deploy_tx_request.go:24]: From is mandatory, why are you using pointer?

File: pkg/types/api/send_tx_request.go [hunk 2]
```diff
@@ -22,8 +23,8 @@ type TransactionParams struct {
 	GasTipCap       string                        `json:"maxPriorityFeePerGas,omitempty" example:"71500000 (wei)"`
 	AccessList      types.AccessList              `json:"accessList,omitempty" swaggertype:"array,object"`
 	TransactionType string                        `json:"transactionType,omitempty" validate:"omitempty,isTransactionType" example:"dynamic_fee" enums:"legacy,dynamic_fee"`
-	From            string                        `json:"from" validate:"omitempty,eth_addr" example:"0x1abae27a0cbfb02945720425d3b80c7e09728534"`
-	To              string                        `json:"to" validate:"required,eth_addr" example:"0x1abae27a0cbfb02945720425d3b80c7e09728534"`
+	From            *ethcommon.Address            `json:"from" validate:"omitempty" example:"0x1abae27a0cbfb02945720425d3b80c7e097285534" swaggertype:"string"`
```
Comment (ggarri) [inline pkg/types/api/send_tx_request.go:26]: From is mandatory, why are you using pointer?

File: pkg/types/api/send_tx_request.go [hunk 3]
```diff
@@ -22,8 +23,8 @@ type TransactionParams struct {
...
+	To              ethcommon.Address             `json:"to" validate:"required" example:"0x1abae27a0cbfb02945720425d3b80c7e09728534" swaggertype:"string"`
```
Comment (ggarri) [inline pkg/types/api/send_tx_request.go:27]: To is nit mandatory, why are you not using pointer?

File: pkg/types/entities/eth_params.go [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 
 import (
+	ethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 )
 
 type ETHTransactionParams struct {
-	From            string               `json:"from,omitempty"  example:"0x1abae27a0cbfb02945720425d3b80c7e09728534"`
-	To              string               `json:"to,omitempty" example:"0x1abae27a0cbfb02945720425d3b80c7e09728534"`
+	From            *ethcommon.Address    `json:"from,omitempty"  example:"0x1abae27a0cbfb02945720425d3b80c7e09728534" swaggertype:"string"`
```
Comment (ggarri) [inline pkg/types/entities/eth_params.go:9]: From is mandatory, why are you using pointer?

File: pkg/types/entities/eth_params.go [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
+	To              ethcommon.Address    `json:"to,omitempty" example:"0x1abae27a0cbfb02945720425d3b80c7e09728534" swaggertype:"string"`
```
Comment (ggarri) [inline pkg/types/entities/eth_params.go:10]: To is not mandatory, why are you not using pointer?

File: pkg/types/entities/faucet.go [hunk 6]
```diff
@@ -1,13 +1,17 @@
...
+	ethcommon "github.com/ethereum/go-ethereum/common"
+)
 
 type Faucet struct {
 	UUID            string
 	Name            string
 	TenantID        string
 	ChainRule       string
-	CreditorAccount string
+	CreditorAccount *ethcommon.Address
```
Comment (ggarri) [inline pkg/types/entities/faucet.go:14]: CreditorAccount is mandatory, why a pointer?

**Review by ggarri — CHANGES_REQUESTED** [review]

**Review by ggarri — APPROVED** [review]

**Pull request merged**